include/zephyr/linker/linker-defs.h

```diff
@@ -48,6 +48,12 @@
 	KEEP(*(SORT_BY_NAME(._##struct_type.static.*))); \
 	_CONCAT(_##struct_type, _list_end) = .
 
+#define Z_LINK_ITERABLE_NUMERIC(struct_type) \
+	_CONCAT(_##struct_type, _list_start) = .; \
+	KEEP(*(SORT(._##struct_type.static.*[0-9]*))); \
+	KEEP(*(SORT(._##struct_type.static.*[1-9][0-9]*))); \
```

Comment:
I don't have a lot of experience with linkers, so ignore if not relevant.

All other defines use `SORT_BY_NAME` explicitly, why not do the same here?
Also, `SORT_BY_INIT_PRIORITY` seems to exist, is this something that could be used instead of a custom numeric implementation?

Reply:
`SORT` is an alias for `SORT_BY_NAME` as per `ld` docs. I just copied other snippets in the same file. I'll try to do some cleanup after (or in parallel) to this PR, so things stay consistent. Regarding `SORT_BY_INIT_PRIORITY`, it says:

> SORT_BY_INIT_PRIORITY is very similar to SORT_BY_NAME. The difference is SORT_BY_INIT_PRIORITY will sort sections into ascending order by numerical value of the GCC init_priority attribute encoded in the section name before placing them in the output file.

ie, `__attribute__ ((init_priority (XXX)))`. I don't think we can use this here.